src/uhttp.c

```diff
@@ -307,9 +309,58 @@ static int process_header_line(const unsigned char* buffer, size_t len, size_t*
     return result;
 }
 
+static void log_data_line(HTTP_CLIENT_HANDLE_DATA* http_data, const char* text_line)
+{
+    char* authStart = strstr(text_line, AUTH_HEADER);
+    char* proxyAuthStart = strstr(text_line, PROXY_AUTH_HEADER);
+    char* authEol = NULL;
+    char* proxyAuthEol = NULL;
+
+    if (authStart != NULL)
+    {
+        authEol = strstr((authStart + 1), HTTP_CRLF_VALUE);
+    }
+    if (proxyAuthStart != NULL)
+    {
+        proxyAuthEol = strstr((proxyAuthStart + 1), HTTP_CRLF_VALUE);
+    }
+    if (authStart == NULL && proxyAuthStart == NULL)
+    {
+        LOG(AZ_LOG_TRACE, LOG_LINE, "%s", text_line);
+    }
+    else if (authStart != NULL && proxyAuthStart == NULL)
+    {
+        if (authEol != NULL)
+        {                                                                 
+            LOG(AZ_LOG_TRACE, LOG_LINE, "%.*s \r\nAuthorization: *** %.*s", (int)(authStart  - text_line), text_line, (int)(strlen(text_line) - (authEol - text_line)), authEol);
+        }
+    }
+    else if (proxyAuthStart != NULL && authStart == NULL)
+    {
+        if (proxyAuthEol != NULL)
+        {
+            LOG(AZ_LOG_TRACE, LOG_LINE, "%.*s \r\nProxy-Authorization: *** %.*s", (int)(proxyAuthStart  - text_line), text_line, (int)(strlen(text_line) - (proxyAuthEol - text_line)), proxyAuthEol);
+        }
+    }
+    else
+    {
+        long int authPos = authStart - text_line;
```

Comment:
Why do you need authPos and proxyAuthPos? Can't you just use authStart and proxyAuthStart in the comparison on line 349?

Reply:
Good point! Fixed.